libkbfs/node_cache_test.go

```diff
@@ -472,3 +473,42 @@ func TestNodeCacheGCReal(t *testing.T) {
 	// Make sure childNode2 isn't GCed until after this point.
 	func(interface{}) {}(childNode2)
 }
+
+type wrappedTestNode struct {
+	Node
+	wrapCalled bool
+}
+
+func (wtn *wrappedTestNode) WrapChild(child Node) Node {
+	child = wtn.Node.WrapChild(child)
+	wtn.wrapCalled = true
+	return child
+}
+
+func TestNodeCacheWrapChild(t *testing.T) {
+	ncs := newNodeCacheStandard(
+		FolderBranch{tlf.FakeID(0, tlf.Private), MasterBranch})
+	var wtn1, wtn2 *wrappedTestNode
+	rw1 := func(root Node) Node {
+		wtn1 = &wrappedTestNode{root, false}
+		return wtn1
+	}
+	rw2 := func(root Node) Node {
+		wtn2 = &wrappedTestNode{root, false}
+		return wtn2
+	}
+	ncs.AddRootWrapper(rw1)
+	ncs.AddRootWrapper(rw2)
+
+	rootPtr := BlockPointer{ID: kbfsblock.FakeID(0)}
+	rootName := "root"
+	rootNode, err := ncs.GetOrCreate(rootPtr, rootName, nil)
+	require.NoError(t, err)
+
+	childPtr := BlockPointer{ID: kbfsblock.FakeID(1)}
+	childName := "child1"
+	_, err = ncs.GetOrCreate(childPtr, childName, rootNode)
```

Comment:
To confirm my understanding, this is where the `WrapChild` gets called on `rootNode`, with the child node passed in as argument, correct?

Reply:
Yes, correct.